dttools/src/set.c

```diff
@@ -174,6 +202,28 @@ int set_insert(struct set *s, const void *element)
 	return 1;
 }
 
+int set_insert_set(struct set *s, struct set *s2)
+{
+	set_first_element(s2);
+	const void *element;
+	while((element = set_next_element(s2))){
+		set_insert(s, element);
+	}
+
+	return 1;
```

Comment:
Here and below you always return 1. Either return void, or the number of elements added.